src/blockydecompression.cpp

```diff
@@ -6,156 +6,84 @@
-{
-	return buffer.size();
-}
-
-void BlockyDecompression::reporter_set_size(size_t size)
-{
-	buffer.resize(size);
-	reporter_position = 0;
-}
-
-std::unique_ptr<DecompressionMethod> BlockyDecompression::get_method_for_block
-(
-	Block block
-)
-{
-	if (!block.HasPattern) return block.HasExponent 
-		? std::unique_ptr<DecompressionMethod>(methods[(int)Methods::FloatSimilar]) 
-			: std::unique_ptr<DecompressionMethod>(methods[(int)Methods::NumbersNoExp]);
-
-	switch (block.Pattern)
-	{
-	case PatternType::Same:
-		return std::unique_ptr<DecompressionMethod>(methods[(int)Methods::PatternSame]);
-	case PatternType::Offset:
-		return std::unique_ptr<DecompressionMethod>(methods[(int)Methods::PatternOffset]);
-	case PatternType::Pingpong:
-		return std::unique_ptr<DecompressionMethod>(methods[(int)Methods::PatternPingPong]);
-	case PatternType::Reserved:
-		// TODO: throw meaningful exception instead of not implemented error code
-		std::cout << "invalid pattern type" << "\n";
-		exit(-501);
-	default:
-		// TODO: throw meaningful exception instead of not implemented error code
-		std::cout << "pattern type not implemented" << "\n";
-		exit(-501);
-	}
+    for (size_t i = 0; i < METHODS_COUNT; i++)
+        delete methods[i];
 }
 
-void BlockyDecompression::initialize(int value_count)
+DecompressionMethod* BlockyDecompression::get_method_for_block(Block block)
 {
-	values.resize(value_count);
+    if (!block.HasPattern) {
+        return block.HasExponent
+            ? (methods[(int)Methods::FloatSimilar])
+            : (methods[(int)Methods::NumbersNoExp]);
+    }
+
+    switch (block.Pattern)
+    {
+    case PatternType::Same:
+        return methods[(int)Methods::PatternSame];
+    case PatternType::Offset:
+        return methods[(int)Methods::PatternOffset];
+    case PatternType::Pingpong:
+        return methods[(int)Methods::PatternPingPong];
+    case PatternType::Reserved:
+        // TODO: throw meaningful exception instead of not implemented error code
+        std::cout << "invalid pattern type" << "\n";
+        exit(-501);
+    default:
+        // TODO: throw meaningful exception instead of not implemented error code
+        std::cout << "pattern type not implemented" << "\n";
+        exit(-501);
+    }
 }
 
 void BlockyDecompression::decompress()
 {
-	size_t value_count = 0;
-	while (value_count < metadata.ValueCount)
-	{
-		if (reader.read_byte(1) > 0) // isBlock
-		{
-			Block block = DecompressionMethod::read_default_block_header(reader, metadata);
-			auto method = get_method_for_block(block);
-			value_count += method->read(reader, block);
-		}
-		else
-		{
-			BlockyNumber value = DecompressionMethod::read_single_value_without_control_bit(reader, metadata);
-			write(value);
-			value_count++;
-		}
-	}
+    size_t value_count = 0;
+    // TODO: lmao this fix just stops the program on segaulting with alpha.water
+    // definitely temporary, remove ASAP
+    if (metadata.ValueCount > 2400) {
```

Comment:
Isn't this already fixed? We did remove the error of having a faulty value count.